mitodl/ocw-studio
# enhancement: upgrade publisher

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/ocw-studio/issues/1821

#### What's this PR do?
Upgrades `ocw-hugo-publisher` to `0.5` (Hugo 0.112.0 and Go 1.20.4). Hugo changed `--baseUrl` to `--baseURL`, so that has been updated too.

#### How should this be manually tested?
1. Navigate to your local ocw-studio project directory.
2. Checkout branch `hussaintaj/1821-upgrade-publisher`
3. Start studio with `docker-compose up`
4. Choose a course you'd like to test this change on.
5. Run
    ```bash
    docker-compose exec web ./manage.py backpopulate_pipelines --filter <course-id>
    docker-compose exec web ./manage.py upsert_theme_assets_pipeline
    docker-compose exec web ./manage.py upsert_mass_build_pipeline
    docker-compose exec web ./manage.py upsert_mass_build_pipeline -o
    ```
6. Open your local concourse http://concourse:8080.
7. Trigger `ocw-theme-assets` build and wait for success.
    **Expected Behavior**: Successful build and image used has the correct tag/version.
    
    <img width="1361" alt="Screenshot 2023-05-31 at 6 29 17 PM" src="https://github.com/mitodl/ocw-studio/assets/71316217/82eb8176-2ba3-4053-a5aa-f1a6e54055dc">
8. Go to your course site on studio UI http://localhost:8043/sites/name
9. Publish the course to live and draft both and wait for success.
    **Expected Behavior**: Successful build and image used has the correct tag/version.
    <img width="1366" alt="Screenshot 2023-05-31 at 6 53 12 PM" src="https://github.com/mitodl/ocw-studio/assets/71316217/b5cc4158-b841-4a47-a711-136e80936e39">
    <img width="1360" alt="Screenshot 2023-05-31 at 6 53 36 PM" src="https://github.com/mitodl/ocw-studio/assets/71316217/da3c551d-b444-4d6a-bcce-ca0326361c4c">
10. Return to the Concourse UI and trigger a mass build. You might want to lower the number of courses that run in the mass build. We can achieve that by setting `publish_date` and `draft_publish_date` to null on all of the courses except the ones you want to test with.
    1. ```bash
        docker-compose exec web ./manage.py shell
        from websites.models import Website
        Website.objects.all().update(publish_date=None, draft_publish_date=None)
        ```
    2. Open https://localhost:8043/admin and set a publish_date so that it appears in the mass build.
    
    **Expected Behavior**: Successful build and image used has the correct tag/version.
    
    <img width="1369" alt="Screenshot 2023-05-31 at 6 55 31 PM" src="https://github.com/mitodl/ocw-studio/assets/71316217/7944f71f-5265-4fb2-ba0c-46d12ff8af81">
